service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesReader.java

```diff
@@ -11,5 +11,9 @@ public interface ResourcesReader {
 
   List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose);
 
+  List<DeployedResource> listResourcesWithPurpose();
+
   List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose);
+
+  List<DeployedVNet> listVNetResourcesWithPurpose();
```

Comment:
I better approach would be to add a new method, listSubnetsByPurpose that returns a new type (DeployedSubnet). This type will contain the subnet name, purpose, region and VNet ID. Once you have this method you can call it for each subnet purpose in the service.

Reply:
As we discussed earlier this week, the subnet has is dependent on VNet context, therefore I'm extracting all VNets that have SubnetResourcePurpose and flatten the structure later in the LZ Service logic. 
That way I'm keeping the model as close as possible to ARM model until very last moment when I have to translate into output formal for service response.

Reply:
The approach of querying all VNets in the RG introduces assumptions we can't guarantee and makes the implementation brittle. In the current test environment, we will find VNets deployed outside the landing zone. In the future we may need to deploy multiple LZs per MRG. Tags are the authoritative way to know if a resource belongs to a LZ. I was under the assumption that we would get the VNets in the RG using the existing public method from the library: `listVNetWithSubnetPurpose`.